### ISSUE TITLE
Add `max` to array field API

### ISSUE THREAD

**Issue by chrisvxd** [measuredco/puck#326]
## Description

Users may wish to limit the number of items added to an array field

## Proposal

A new `max` API for the array field type

```js
const config = {
  components: {
    Example: {
      fields: {
        items: {
          type: "array",
          arrayFields: {
            title: { type: "text" },
          },
          max: 5 // new
        },
      },
  },
};
```

**Comment by xaviemirmon** [comment]
@chrisvxd love this! +1 from me.  Could we also add a min to this issue?

```
const config = {
  components: {
    Example: {
      fields: {
        items: {
          type: "array",
          arrayFields: {
            title: { type: "text" },
          },
          max: 5 // new
          min: 2 // new
        },
      },
  },
};
```

**Comment by chrisvxd** [comment]
Implemented by #341

**Comment by xaviemirmon** [comment]
I am just playing around with the demo on this @chrisvxd.  I feel like the plus button should be hidden when you reach max and the delete button when you reach min.  Should this be re-opened to include this or a new issue raised?
